ArduCopter/mode_guided.cpp

```diff
@@ -1177,4 +1238,61 @@ uint32_t ModeGuided::get_timeout_ms() const
     return MAX(copter.g2.guided_timeout, 0.1) * 1000;
 }
 
+/********************************************************************************/
+// Pause and continue guided mode
+/********************************************************************************/
+
+// pause guide mode
+bool ModeGuided::pause()
+{
+    if(init(true)) {
+        copter.prev_cmd.paused = true;
+        return true;
+    }
+    copter.prev_cmd.paused = false;
+    return false;
+}
+
+// resume guided mode
+bool ModeGuided::resume()
+{
+    // check if vehicle paused to not re-enter resume
+    if (!copter.prev_cmd.paused) {
+        return false;
+    }
+    copter.prev_cmd.paused = false;
+
+    // resume to previous command
+    switch (copter.prev_cmd.prev_cmd_type) {
+        case Copter::PrevCmd::DO_USER_TAKEOFF_START: {
+            return do_user_takeoff_start(copter.prev_cmd.takeoff_alt_cm);
+        }
+        case Copter::PrevCmd::SET_DESTINATION_VECTOR: {
+            return set_destination(copter.prev_cmd.destination, copter.prev_cmd.use_yaw, copter.prev_cmd.yaw_cd,
+                                   copter.prev_cmd.use_yaw_rate, copter.prev_cmd.yaw_rate_cds,
+                                   copter.prev_cmd.relative_yaw, copter.prev_cmd.terrain_alt);
+        }
+        case Copter::PrevCmd::SET_DESTINATION_LOCATION: {
+            return set_destination(copter.prev_cmd.dest_loc, copter.prev_cmd.use_yaw, copter.prev_cmd.yaw_cd,
+                                   copter.prev_cmd.use_yaw_rate, copter.prev_cmd.yaw_rate_cds,
+                                   copter.prev_cmd.relative_yaw);
+        }
+        case Copter::PrevCmd::SET_DESTINATION_POSVEL: {
+            return set_destination_posvel(copter.prev_cmd.destination, copter.prev_cmd.velocity,
+                                          copter.prev_cmd.use_yaw, copter.prev_cmd.yaw_cd, copter.prev_cmd.use_yaw_rate,
+                                          copter.prev_cmd.yaw_rate_cds, copter.prev_cmd.relative_yaw);
+        }
+        case Copter::PrevCmd::SET_DESTINATION_POSVELACCEL: {
+            return set_destination_posvelaccel(copter.prev_cmd.destination, copter.prev_cmd.velocity,
+                                               copter.prev_cmd.acceleration, copter.prev_cmd.use_yaw,
+                                               copter.prev_cmd.yaw_cd, copter.prev_cmd.use_yaw_rate,
+                                               copter.prev_cmd.yaw_rate_cds, copter.prev_cmd.relative_yaw);
+        }
+        default: {
+            return false;
+        }
```

Comment:
```suggestion
```

Omitting this forces someone who adds a PrevCmd entry to deal with the continue case.  Generally when a value is constrained to being within an enumeration then we omit the `default` clause.